course_catalog/etl/enrollments.py

```diff
@@ -0,0 +1,78 @@
+"""Course run enrollments"""
+import pyathena
+import pytz
+from pyathena.cursor import DictCursor
+from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
+from course_catalog.models import LearningResourceRun, Course, Enrollment
+
+
+def athena_cursor():
+    """Returns authorized pyathena cursor"""
+    return pyathena.connect(
+        aws_access_key_id=settings.ATHENA_ACCESS_KEY_ID,
+        aws_secret_access_key=settings.ATHENA_SECRET_ACCESS_KEY,
+        work_group=settings.ATHENA_WORK_GROUP,
+        cursor_class=DictCursor,
+        region_name=settings.ATHENA_REGION_NAME,
+    ).cursor()
+
+
+def update_enrollments_for_user(user):
+    """
+    Updates enrollment data for a singe user
+
+    Args:
+        user (User): user to fetch enrollments for
+    """
+
+    if not (
+        settings.ATHENA_ACCESS_KEY_ID
+        and settings.ATHENA_SECRET_ACCESS_KEY
+        and settings.ATHENA_WORK_GROUP
+        and settings.ATHENA_REGION_NAME
+    ):
+        return
+
+    query = """
+        SELECT 
+            enrollment_course_id,
+            enrollment_created
+        FROM ol_warehouse.mitx_enrollments 
+        WHERE email = '{email}'
+        """.format(
+        email=user.email
+    )
+
+    cursor = athena_cursor()
+    cursor.execute(query)
+
+    for result in cursor:
+        run_id = result["enrollment_course_id"]
+        run_id_parts = run_id.split("/")
+
+        run_regex_string = ".*{course_code}.*{run_code}.*".format(
+            course_code=run_id_parts[1], run_code=run_id_parts[2]
+        )
+        run = LearningResourceRun.objects.filter(
+            run_id__regex=run_regex_string,
+            content_type=ContentType.objects.get(model="course").id,
+            published=True,
+        ).last()
+
+        if run:
+            course = run.content_object
+        else:
+            course = Course.objects.filter(
+                course_id__contains=run_id_parts[1], published=True
+            ).last()
+
+        timezone = pytz.timezone("UTC")
+        enrollment_timestamp = timezone.localize(result["enrollment_created"])
+
+        Enrollment.objects.update_or_create(
```

Comment:
Is it okay to create an enrollment if both `run` and `course` are `None`?  Should there be an error logged if `course` is also `None` to notify someone that the Athena `enrollment_course_id` doesn't have any match in open?

Reply:
There are enough courses without matches that raising an error does not make sense. Once this is merged and the data is updated, the plan is to review the enrollment_course_ids with no matches to see if there are any we care about then add a look up table to match them to the correct course id